Pitch Solo Extraction

ewega opened the pull request:
- Created sendPitch.py , gets only pitch from youtube URL
- Also edited import-clip code to run locally easliy (when testing)
- ALSO added progress prompts to main.py (if we ever even use it)

**TESTING**
We want to test locally, so for a random video (ideally kind of short just for testing) do the following. DJ Trump Video: "_https://www.youtube.com/watch?v=JZRXESV3R74_"

- Turn on localhost and everything
- Create "_**labelled**_" table locally (you might have to create the columns too idk)
- Add a "pitch" column to **_"labelled"_** table
- cd `import-clips`
- `npm start https://www.youtube.com/watch?v=JZRXESV3R74 log`
This should fill the local table with empty rows for all time intervals.

- cd `audio-analysis`
- `python sendPitch.py log`
- Enter the DJ Link above (put whatever for tag)
When finished running, the "pitch" column for that URL in **_"labelled"_** table should be filled out.

Might have dependencies and stuff.

NEXT STEPS: Run this code on our prod DB.

![image](https://user-images.githubusercontent.com/26189114/75752888-95e96580-5cf7-11ea-9e38-0ec8979dcd10.png)

ewega committed:
Pitch Solo Extraction

- Created sendPitch.py , gets only pitch from youtube URL
- Also edited import-clip code to run locally easliy (when testing)
- ALSO added progress prompts to main.py (if we ever even use it)

tjlam commented on audio-analysis/lib/etl/dfHelper.py:
```diff
@@ -24,3 +24,17 @@ def customLeftJoin(dfLeft, dfRight, joinCondition):
+            """
+    engine.execute(sql)
+    sql = "DROP TABLE `temp_table`"
+    engine.execute(sql)
```
new line

tjlam commented on audio-analysis/lib/etl/sqlConnection.py:
```diff
@@ -44,11 +44,20 @@ def sqlConnectionSetup(engine):
+#Function checks if url exists in table already
+def urlExistsSpecific(url, engine, column_name):
+    # Check if URL already exists
+    result = engine.execute('SELECT * FROM `labelled` WHERE `url`="' + url + '" AND `pitch` IS NOT NULL')
```
Should we pass in `column_name` into the sql query? or is it only used for the print statement?

tjlam replied:
Why do we check if pitch is not null? If it is null we still want to update it right?

tjlam commented on audio-analysis/sendPitch.py:
```diff
@@ -0,0 +1,42 @@
+engine = getEngine(TESTING)
+
+url = input("Provide YT Link: ")
+tag = input("Provide highlight tag (True/False): ")
```
Do we need this anymore?

tjlam commented on audio-analysis/sendPitch.py:
```diff
@@ -0,0 +1,42 @@
+
+engine = getEngine(TESTING)
+
+url = input("Provide YT Link: ")
```
Can we remove input prompts. I find its faster to add the yt url in the args. `sendPitch.py [yt url] log`

tjlam commented on import-clips/run.js:
```diff
@@ -60,14 +74,29 @@ const OVERLAP_TIME = 2;
-    sendToDB(newRows);
+    newRows = clipIntervals.map( interval => formatRow(youtubeUrl, interval));
+    if(TESTING){
+      con.connect(function(err) {
```
I should have done it this way to start but I was lazy and used the flask app end point. Let's make this into a method, and we can use it to update local or prod.

tjlam reviewed (commented):
Looks good, mostly some nits and questions

ewega committed:
Lil fixes here and there

Eldrick19 commented:
Made the fixes.

tjlam commented on audio-analysis/lib/utils/videoHelper.py:
```diff
@@ -67,7 +66,8 @@ def getFilename(self):
         video_title = self.info_dict.get('title', None)
-        filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
+        #filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
+        filename_wav = 'audio-files/CS -GO - Fnatic vs. mousesports [Inferno] Map 1 - Grand Final - ESL Pro League Season 10 Finals-0AlFcix7aaY.wav'
```
is this a test file?

mattlianje commented on audio-analysis/lib/etl/sqlConnection.py:
```diff
@@ -44,11 +44,20 @@ def sqlConnectionSetup(engine):
     meta.create_all() # Create all tables
 
+#Function checks if url exists in table already
+def urlExistsSpecific(url, engine, column_name):
```
helpful!

tjlam commented on audio-analysis/lib/etl/sqlConnection.py:
```diff
@@ -44,11 +44,20 @@ def sqlConnectionSetup(engine):
+#Function checks if url exists in table already
+def urlExistsSpecific(url, engine, column_name):
+    # Check if URL already exists
+    result = engine.execute('SELECT * FROM `labelled` WHERE `url`="' + url + '" AND `'+ column_name +'` IS NOT NULL')
```
I still don't understand the `column_name IS NOT NULL` part of the query. Why are we filtering by not null? If we import a video by default the pitch would be null so this method would return false

mattlianje replied:
Aaaaaa makes sense. @tjlam @Eldrick19 expained it to me. Basically it just checks if `column_name` has been filled in for a given `url` but the name of the method is a bit exotic.

tjlam replied:
Why do we need to check if that column is filled though? If this returns false we don't want to update right?

ewega replied:
@tjlam You're right. I think it helps looking at this part of the code:
```
url_exists = urlExistsSpecific(url, engine, column_name)
if url_exists == True:
    print("This video is already in the database. Will overwrite ", column_name, " data.")
```

It should say "This video's pitch is already in the database" or whatever - mb on wording there. But essentially its just for logging.

tjlam replied:
Let's say that url is not in the database at all, would `updateDBWithDataFromDF` still work?

tjlam replied:
I assume it wouldn't??? because it would have to insert new rows

ewega replied:
That is correct, it assumes the URL is already in the DB

mattlianje commented on audio-analysis/lib/utils/videoHelper.py:
```diff
@@ -67,7 +66,8 @@ def getFilename(self):
     def getFilenameWav(self):
         video_title = self.info_dict.get('title', None)
-        filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
+        #filename_wav = filepath + 'WAV-' + video_title + '-' + self.url.split("=", 1)[1] + '.wav'
```
let s avoid wildcards and have this for all audio files

mattlianje reviewed (changes requested):
LGTM) one lil one)

ewega committed:
Fixed getFilenameWav method

tjlam commented:
You can add your secrets file to the import-clips folder. Just make sure you update `.gitignore` with `import-clips/secretsJS.js`

ewega committed:
Few fixes

- Added urlExists check to sendPitch.py
- Send urlExists to check in labelled table and not test_table
- Renamed methods for clarity

ewega committed:
Deleted Files

mattlianje reviewed (approved):
`bang`
